firmware/library/L1_Drivers/dac_test.cpp

```diff
@@ -0,0 +1,110 @@
+// Test for Dac class.
+// Using a test by side effect on the LPC_IOCON register
+#include "L1_Drivers/dac.hpp"
+#include "L1_Drivers/pin_configure.hpp"
+#include "L5_Testing/testing_frameworks.hpp"
+// Start of Unit Tests for the DAC
+TEST_CASE("Testing Dac", "[dac]")
+{
+    using fakeit::Fake;
+    using fakeit::Mock;
+    using fakeit::Spy;
+    using fakeit::Verify;
+    using fakeit::When;
+    using fakeit::Using;
+
+    LPC_IOCON_TypeDef local_iocon;
+    memset(&local_iocon, 0, sizeof(local_iocon));
+    LPC_DAC_TypeDef local_dac_port;
+    Dac::dac_register = &local_dac_port;
+    PinConfigure iocon_pin_config(0, 26);
+    PinConfigure::pin_map =
+        reinterpret_cast<PinConfigure::PinMap_t *>(&local_iocon);
+    PinConfigure test_pin(0, 26);
```

Comment:
Pin configure shouldn't be tested via side effect. It should be tested soley though Mock using fakeit::Mock.